src/preact.d.ts

```diff
@@ -101,17 +101,18 @@ declare namespace preact {
 		abstract render(props?: RenderableProps<P>, state?: Readonly<S>, context?: any): ComponentChild;
 	}
 
-	function h<P>(
-		node: ComponentFactory<P>,
-		params: Attributes & P | null,
-		...children: ComponentChildren[]
-	): VNode<any>;
 	function h(
 		node: string,
 		params: JSX.HTMLAttributes & JSX.SVGAttributes & Record<string, any> | null,
 		...children: ComponentChildren[]
 	): VNode<any>;
 
+	function h<P>(
+		node: ComponentFactory<P>,
+		params: Attributes & P & Record<string, any> | null,
```

Comment:
Were you running into an issue with the `props` typings with the `Records<string, any>`?

Reply:
Checking again, we actually don't need this change